packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -961,8 +961,13 @@ export class Cluster extends ClusterBase {
 
     this.vpcSubnets = props.vpcSubnets ?? [{ subnetType: ec2.SubnetType.PUBLIC }, { subnetType: ec2.SubnetType.PRIVATE }];
 
+    const selectedSubnetIds = this.vpcSubnets.map(s => this.vpc.selectSubnets(s).subnetIds);
```

Comment:
```suggestion
    const selectedSubnetIdsPerGroup = this.vpcSubnets.map(s => this.vpc.selectSubnets(s).subnetIds);
```

Or something, the current name makes it sound like these are individual subnets.